listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -695,6 +695,36 @@ def delete(self, musicbrainz_id):
             self.log.error("Cannot delete listens for user: %s" % str(e))
             raise
 
+    def delete_listen(self, musicbrainz_id:str, recording_msid: str):
+        """ Delete a particular listen for user with specified MusicBrainz ID.
+
+        Args:
+            musicbrainz_id: the MusicBrainz ID of the user
+            recording_msid: the MessyBrainz ID of the recording
+
+        Raises: Exception if unable to delete the listen
+        """
+
+        args = {'user_name': musicbrainz_id, 'recording_msid': recording_msid}
+        query = """DELETE FROM listen
+                    WHERE user_name = :user_name
+                      AND recording_msid = :recording_msid"""
```

Comment:
this would delete ALL listens with a particular recording msid, is that what we want to do? If so, we should fix the name of the function to make sure that it's clear that it will probably delete more than one listen.

Reply:
I changed it to delete the listen based on (listened_at, track_name, user_name) since the `listen` table in timescale is now based on track_name rather than recording_feedback as a key. Hopefully, your issue is addressed now.

Reply:
Why not use recording msid instead of track name?

Reply:
+1

Reply:
Reverted